src/com/amazon/corretto/crypto/provider/HkdfRfc5869Spec.java

```diff
@@ -0,0 +1,152 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.corretto.crypto.provider;
+
+import java.security.spec.KeySpec;
+
+public class HkdfRfc5869Spec implements KeySpec {
+  final int mode;
+  final byte[] secretOrPrk;
+  final byte[] salt;
+  final byte[] info;
+  final int desiredLength;
+  final String algorithmName;
+  public static final String DEFAULT_ALGORITHM_NAME = "HkdfRfc5869";
+  public static final int HKDF_MODE = 1;
+  public static final int HKDF_EXTRACT_MODE = 2;
+  public static final int HKDF_EXPAND_MODE = 3;
```

Comment:
Why not an Enum? Is this a JCA convention?

Reply:
The Cipher class uses int for modes of operation and I'm just following that convention.